File: testdata/jennies/builders/struct_fields_as_args_assignment/CSharpBuilder/src/Grafana/Foundation/Sandbox/SomeStructBuilder.cs
namespace Grafana.Foundation.Sandbox;


public class SomeStructBuilder : Cog.IBuilder<SomeStruct>
{
    protected readonly SomeStruct @internal;

    public SomeStructBuilder()
    {
        this.@internal = new SomeStruct();
    }

    public SomeStructBuilder Time(string from,string to)
    {
        if (this.@internal.Time == null)
        {
            this.@internal.Time = new object();
        }
        this.@internal.Time.From = from;
        this.@internal.Time.To = to;
        return this;
    }

    public SomeStruct Build()
    {
        return this.@internal;
    }
}

File: testdata/jennies/builders/deprecated_builder/JavaBuilder/sandbox/SomeStructBuilder.java
package sandbox;


/**
 * @deprecated This builder is deprecated. Don't use. Please.
 */
@Deprecated(forRemoval = true)
public class SomeStructBuilder implements cog.Builder<SomeStruct> {
    protected final SomeStruct internal;
    
    public SomeStructBuilder() {
        this.internal = new SomeStruct();
    }
    public SomeStructBuilder title(String title) {
        this.internal.title = title;
        return this;
    }
    public SomeStruct build() {
        return this.internal;
    }
}
